lib/Dialect/TritonNvidiaGPU/Transforms/MaterializeLoadStore.cpp

```diff
@@ -63,34 +63,49 @@ struct MaterializeLoadStorePass
   }
 
   void runOnOperation() override {
-    SmallVector<mlir::triton::LoadOp> worklists;
-    getOperation()->walk([&](mlir::triton::LoadOp load) -> void {
-      if (isLoadFromTensorPtr(load)) {
-        worklists.push_back(load);
-      }
+    Block *entryBlock;
+    getOperation()->walk([&](mlir::triton::FuncOp funcOp) {
+      entryBlock = &funcOp.getBody().front();
     });
-    for (auto load : worklists) {
-      materializeLoadTilePtr(load);
+
+    SmallVector<mlir::triton::LoadOp> loadOpWorkList;
+    getOperation()->walk<WalkOrder::PostOrder>(
+        [&](mlir::triton::LoadOp load) -> void {
```

Comment:
nit:  "-> void" can be omitted.

and, just curious, why do you intentionally set it to <WalkOrder::PostOrder>? I feel the algo does not have requirements on this?

Reply:
`-> void` has been deleted.
 Actually, this algo does not need this requirement:`WalkOrder::PostOrder`, I have deleted it.